features/authentication.feature

```diff
@@ -117,6 +117,45 @@ Feature: OAuth authentication
     Then the output should not contain "github.com password for mislav"
     And the file "../home/.config/hub" should contain "oauth_token: OTOKEN"
 
+  Scenario: Credentials from GITHUB_USER & GITHUB_TOKEN
+    Given the GitHub API server:
+      """
+      post('/authorizations') {
+        assert_basic_auth 'mislav', 'kitty'
+        json :token => 'OTOKEN'
+      }
+      get('/user') {
+        json :login => 'mislav'
+      }
+      post('/user/repos') {
+        json :full_name => 'mislav/dotfiles'
+      }
+      """
+    Given $GITHUB_USER is "mislav"
+    And $GITHUB_TOKEN is "feline0123"
+    When I successfully run `hub create`
+    Then the output should not contain "github.com password for mislav"
+    And the file "../home/.config/hub" should contain "oauth_token: feline0123"
+
+  Scenario: GITHUB_TOKEN without user
+    Given the GitHub API server:
+      """
+      post('/authorizations') {
+        assert_basic_auth 'mislav', 'kitty'
+      }
+      get('/user') {
+        json :login => 'mislav'
+      }
+      post('/user/repos') {
+        json :full_name => 'mislav/dotfiles'
+      }
+      """
+    Given $GITHUB_TOKEN is "feline01234"
+    When I run `hub create` interactively
+    When I type "mislav"
```

Comment:
Ah, ok. From your comment ( https://github.com/github/hub/issues/463#issuecomment-32387755 ) I thought that the username was required. I'll rework this at your request, however!